BoneLib/BoneLib/HelperMethods.cs

```diff
@@ -132,13 +141,26 @@ public static byte[] GetResourceBytes(Assembly assembly, string name)
             }
             return null;
         }
-        
+
         ///<summary>
         /// Checks if an assembly is loaded from name
         /// </summary>
         public static bool CheckIfAssemblyLoaded(string name)
         {
-            return AppDomain.CurrentDomain.GetAssemblies().Any(asm => asm.GetName().Name.ToLower().Contains(name.ToLower()));
+            HashSet<Assembly> query = new HashSet<Assembly>(AppDomain.CurrentDomain.GetAssemblies());
+            bool found = false;
+
+            foreach (var queryAsm in query)
+            {
+                string asmName = queryAsm.GetName().Name;
+                if (asmName.ToLower() == name.ToLower())
+                {
+                    found = true;
+                    break;
```

Comment:
nit: could be replaced by just returning `true` immediately, and then returning false at the bottom